pkg/controller/master/rancher/namespace.go

```diff
@@ -0,0 +1,104 @@
+package rancher
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sschema "k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/harvester/harvester/pkg/settings"
+	"github.com/harvester/harvester/pkg/util"
+)
+
+func (h *Handler) onNamespaceRemoved(_ string, namespace *corev1.Namespace) (*corev1.Namespace, error) {
+	if namespace == nil {
+		return namespace, nil
+	}
+
```

Comment:
allow additional label/annotation to skip the cascade-deleting quickly on L22? in case error blocking

Reply:
use a new annotation here, the main consideration is:

if remote Rancher is down/not reachable, then delete any namespace on Harvester is blocked, the current solution is to disable the `settings`

if we allow certain namespace to skip the check, it will be another solution for abovce case

it depends on your choice to add it or not